src/lib.rs

```diff
@@ -93,6 +93,17 @@ impl Module {
         }
     }
 
+    /// Deserialize a module from binary form.
+    pub fn safe_read(wasm_buf: &[u8]) -> Result<Module, ()> {
```

Comment:
@pepyakin do you prefer `Option`, this or `Result` with a proper error message?

Reply:
Also alternatively this could just replace `read` if you agree there's no need for the panicing version.

Reply:
Yes, please! I don't think we need aborting, `BinaryenModuleSafeRead` is simply superior!

I think `Result<_, ()>` works! We can add a message later on.